sample_factory_examples/dmlab/dmlab_level_cache.py

```diff
@@ -218,9 +218,9 @@ def add_new_level(self, level, seed, key, pk3_path):
 def dmlab_ensure_global_cache_initialized(experiment_dir, all_levels_for_experiment, num_policies, level_cache_dir):
     global DMLAB_GLOBAL_LEVEL_CACHE
 
-    assert (
-        multiprocessing.current_process().name == "MainProcess"
-    ), "make sure you initialize DMLab cache before child processes are forked"
+    # assert (
+    #     multiprocessing.current_process().name == "MainProcess"
+    # ), "make sure you initialize DMLab cache before child processes are forked"
```

Comment:
There should be a way to propagate cache across processes, i.e. via sf_global_context()
So I believe this assert is still relevant?

Reply:
Whenever I ran the example this assert would be triggered. I will take a look into it.

Reply:
The thing is in SF2 we can't just store data in global variables. If we want to use global state it should be communicated through sf_global_context() because this context is passed on explicitly when we create new processes.
Or we should use a different mechanism that does not rely on global state, not sure what is better.

Cache mechanism currently uses a bunch of global state I believe and therefore needs to be reworked. Probably initializing it explicitly and then passing along through global context is the easiest way to do this :)

This all worked in SF1 because it used "fork" multiprocessing method, but this created lots of other limitations, such as not being able to use CUDA in the main process and problems with envs like isaacgym.